PWGMM/Tasks/uecharged.cxx

```diff
@@ -0,0 +1,336 @@
+    if (TMath::Abs(collision.posZ()) > 10.0)
+      return;
+    auto vtxZ = collision.posZ();
+
+    ue.fill(HIST("hCounter"), 2);
+
+    ue.fill(HIST("hvtxZ"), vtxZ);
+    // loop over selected tracks
+    double flPt = 0; // leading pT
+    double flPhi = 0;
+    int flIndex = 0;
+    for (auto& track : tracks) {
+
+      if (!globalTracks.IsSelected(track))
+        continue;
+
+      ue.fill(HIST("hdNdeta"), track.eta());
+      ue.fill(HIST("vtxZEta"), track.eta(), vtxZ);
+      ue.fill(HIST("phiEta"), track.eta(), track.phi());
+
+      if (track.pt() < 0.15)
+        continue;
+      if (flPt < track.pt()) {
+        flPt = track.pt();
+        flPhi = track.phi();
+        flIndex = track.globalIndex();
+      }
+    }
+    ue.fill(HIST("hPtLeadingRecPS"), flPt);
+    vector<double> ue_rec;
+    ue_rec.clear();
+    int nchm_top[3];
+    double sumptm_top[3];
+    for (int i = 0; i < 3; ++i) {
+      nchm_top[i] = 0;
+      sumptm_top[i] = 0;
+    }
+    //LOGF (INFO, "--------------------   FLAG 0!!!!!!!!!!!");
+    vector<Float_t> ptArray;
+    vector<Float_t> phiArray;
+    vector<int> indexArray;
+    ptArray.clear();
+    phiArray.clear();
+    indexArray.clear();
+
+    for (auto& track : tracks) {
+
+      if (track.pt() < 0.15)
+        continue;
+
+      if (globalTrackswoPrim.IsSelected(track))
+        ue.fill(HIST("hPTVsDCAData"), track.pt(), track.dcaXY());
+
+      if (!globalTracks.IsSelected(track))
+        continue;
+
+      // applying the efficiency twice for the misrec of leading particle
+      if (f_Eff->Eval(track.pt()) > gRandom->Uniform(0, 1)) {
+        ptArray.push_back(track.pt());
+        phiArray.push_back(track.phi());
+        indexArray.push_back(track.globalIndex());
+      }
+
+      // remove the autocorrelation
+      if (flIndex == track.globalIndex())
+        continue;
+
+      double DPhi = DeltaPhi(track.phi(), flPhi);
+
+      // definition of the topological regions
+      if (TMath::Abs(DPhi) < M_PI / 3.0) { // near side
+        ue.fill(HIST(hPhi[0]), DPhi);
+        ue.fill(HIST(hPtVsPtLeadingData[0]), flPt, track.pt());
+        nchm_top[0]++;
+        sumptm_top[0] += track.pt();
+      } else if (TMath::Abs(DPhi - M_PI) < M_PI / 3.0) { // away side
+        ue.fill(HIST(hPhi[1]), DPhi);
+        ue.fill(HIST(hPtVsPtLeadingData[1]), flPt, track.pt());
+        nchm_top[1]++;
+        sumptm_top[1] += track.pt();
+      } else { // transverse side
+        ue.fill(HIST(hPhi[2]), DPhi);
+        ue.fill(HIST(hPtVsPtLeadingData[2]), flPt, track.pt());
+        nchm_top[2]++;
+        sumptm_top[2] += track.pt();
+      }
+    }
+
+    for (int i_reg = 0; i_reg < 3; ++i_reg)
+      ue_rec.push_back(1.0 * nchm_top[i_reg]);
+    for (int i_reg = 0; i_reg < 3; ++i_reg)
+      ue_rec.push_back(sumptm_top[i_reg]);
+
+    // add flags for Vtx, PS, ev sel
+
+    // for some reason loops in side donot work
+    ue.fill(HIST(pNumDenMeasuredPS[0]), flPt, ue_rec[0]);
```

Comment:
`HIST()` macro requires literal constant as an argument. To use loop here you may want to retrieve pointers to the histograms from the registry into a vector.

Reply:
Hi, 
sorry for jumping in here. Just as a small comment to this:
An alternative to working with the pointers as Anton suggested would be to use the `static_for` loop that is specifically dedicated to this purpose 
(see e.g. here: https://github.com/AliceO2Group/O2Physics/blob/master/Tutorials/src/histogramRegistry.cxx#L227-L236 
or 
https://github.com/AliceO2Group/AliceO2/blob/dev/Framework/Core/test/test_StaticFor.cxx
).
Then it would be something like
```
#include "Framework/StaticFor.h"

static_for<0, 2>([&](auto i) {
    constexpr int index = i.value;
   ue.fill(HIST(pNumDenMeasuredPS[index]), flPt, ue_rec[index]);  
   ue.fill(HIST(pNumDenData[index]), flPt, ue_rec[index]);
   ue.fill(HIST(pSumPtMeasuredPS[index]), flPt, ue_rec[index+3]);
   ue.fill(HIST(pSumPtData[index]), flPt, ue_rec[index+3]);
});
```
Cheers,
  Mario

Reply:
This is even better, as it avoids dealing with pointers. Thank you, Mario!

Reply:
Thanks @mario-krueger, it works very well.